src/java/htsjdk/variant/variantcontext/CommonInfo.java

```diff
@@ -166,29 +189,33 @@ public void setLog10PError(double log10PError) {
     // Working with attributes
     //
     // ---------------------------------------------------------------------------------------------------------
+    /** removes all attributes */
     public void clearAttributes() {
         attributes = new HashMap<String, Object>();
     }
 
     /**
+     * return an unmodifiable Map of attributes
      * @return the attribute map
      */
     public Map<String, Object> getAttributes() {
         return Collections.unmodifiableMap(attributes);
     }
 
     // todo -- define common attributes as enum
-
-    public void setAttributes(Map<String, ?> map) {
+    /** clear all attributes, and set 'map' as attributes */
+    public void setAttributes(final Map<String, ?> map) {
```

Comment:
Document whether subsequent changes to the `map` passed in will be reflected in this record (they won't).